src/Encamina.Enmarcha.SemanticKernel/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -29,6 +30,25 @@ public static IServiceCollection AddMemoryManager(this IServiceCollection servic
         return services;
     }
 
+    /// <summary>
+    /// Adds and configures the <see cref="MemoryManager"/> type as transient service instance of the <see cref="IMemoryManager"/> service to the <see cref="IServiceCollection"/>.
+    /// </summary>
+    /// <remarks>
+    /// This method registers the <see cref="MemoryManager"/> type as a transient service due to its dependency on <see cref="Kernel"/>, which is typically registered as transient or scoped (rarely as singleton).
+    /// </remarks>
+    /// <param name="services">The <see cref="IServiceCollection"/> to add services to.</param>
+    /// <param name="memoryStorageEventHandlerDelegate">A delegate to handle events from <see cref="IMemoryManager.MemoryStorageEvent"/>.</param>
+    /// <returns>The <see cref="IServiceCollection"/> so that additional calls can be chained.</returns>
+    public static IServiceCollection AddMemoryManager(
+        this IServiceCollection services,
+        Action<string, MemoryStorageEventArgs> memoryStorageEventHandlerDelegate)
```

Comment:
Idem. People do not use the `delegate` suffix anymore. Just call it `memoryStorageEventHandler`. The fact that it is an `Action` already indicates it is a delegate.

Reply:
Right, you're right, I'm renaming it.